luarules/gadgets/api_unit_suspension_handler.lua

```diff
@@ -0,0 +1,250 @@
+local gadget = gadget ---@type Gadget
+
+function gadget:GetInfo()
+    return {
+        name    = "Suspended Unit Handler",
+        desc    = "Prevent actions by units that are stunned, trapped, or disabled.",
+        author  = "efrec",
+        date    = "2025",
+        version = 0,
+        license = "GNU GPL, v2 or later",
+        layer   = -999999,
```

Comment:
why -999999?

Reply:
Sounds fine. 0 is the "I don't intend this to have any specific priority" default but if it's intended to run first then -999999 is correct afaict.